pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -306,7 +306,25 @@ class ModelParser {
                             break
                         case 'steps':
                             def stepsBlock = matchBlockStatement(s);
-                            stage.branches.addAll(parseStepsBlock(asBlock(stepsBlock.body.code)))
+                            BlockStatement block = asBlock(stepsBlock.body.code)
```

Comment:
Why are you inlining this block? didn't we have the code creep issue refactored once already?